scripts/_common.sh

```diff
@@ -130,3 +126,60 @@ function use_gateway_binaries_for_version() {
 }
 export -f use_gateway_binaries_for_version
 
+function generate_matrix() {
+  filter_fn="$1"
+  shift
+
+  versions=("$@")
+  version_matrix=()
+  for fed_version in "${versions[@]}"; do
+    for client_version in "${versions[@]}"; do
+      for gateway_version in "${versions[@]}"; do
+        if "$filter_fn" "$fed_version" "$client_version" "$gateway_version"; then
+          continue
+        fi
+
+        version_matrix+=("$fed_version $client_version $gateway_version")
```

Comment:
Nit: Seems like we could echo it here right away.

Reply:
Good call 👍 

I also changed the filter logic to be inclusive instead of a `filter_not`, since that's a bit of a mind twister.